chain/index/indexer.go

```diff
@@ -21,6 +21,7 @@ var _ Indexer = (*SqliteIndexer)(nil)
 
 // IdToRobustAddrFunc is a function type that resolves an actor ID to a robust address
 type IdToRobustAddrFunc func(ctx context.Context, emitter abi.ActorID, ts *types.TipSet) (address.Address, bool)
+type tipsetExecutorFnc func(ctx context.Context, ts *types.TipSet) (cid.Cid, cid.Cid, error)
```

Comment:
Leaky abstraction here, since the two CIDs are always discarded. A couple of options to make this a bit nicer:

1. Just use an `error` return and make the wrapper function 1 line longer to discard the others (even though currently it doesn't even need to be a wrapper, just `indexer.SetTipsetExecutorFnc(sm.RecomputeTipSetState)` would do).
2. Just call it `recomputeTipSetState` and admit that this is all we're expecting here.
3. Make `SqliteIndexer` depend on `StateManager` and load it in via fx to avoid jumping through hoops and to be completely honest.


(also, `Fnc` -> `Func`?)

Reply:
@rvagg Fixed this by using a  `recomputeTipSetState` function here that simply returns an error and passing in a wrapper to hide the AMT cids during construction.